sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/python/PythonDataSourceV2.scala

```diff
@@ -52,8 +52,22 @@ class PythonDataSourceV2 extends TableProvider {
     dataSourceInPython
   }
 
-  def setDataSourceInPython(dataSourceInPython: PythonDataSourceCreationResult): Unit = {
-    this.dataSourceInPython = dataSourceInPython
+  private var readInfo: PythonDataSourceReadInfo = _
+  def getOrCreateReadInfo(
+      shortName: String,
+      options: CaseInsensitiveStringMap,
+      outputSchema: StructType,
+      isStreaming: Boolean
+  ): PythonDataSourceReadInfo = {
```

Comment:
does this method need to be thread-safe?

Reply:
I'm fairly sure that it doesn't need to be thread safe since `getOrCreateDataSourceInPython` is also non thread-safe.
https://github.com/apache/spark/blob/9dd35b5cbd3ab8822731262659ba84bd1a4726d3/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/python/PythonDataSourceV2.scala#L45-L53

From my quick searches online it seems like catalyst is sequential per query?